src/DS/sds.c

```diff
@@ -255,16 +266,11 @@ int ds_sds_dump_component_ref_as(xmlNodePtr component_ref, struct ds_sds_session
 	char* filename_cpy = oscap_sprintf("./%s", relative_filepath);
 	char* file_reldir = dirname(filename_cpy);
 
-	// the cast is safe to do because we are using the GNU basename, it doesn't
-	// modify the string
-	const char* file_basename = basename((char*)relative_filepath);
-
-	const char* target_filename_dirname = oscap_sprintf("%s/%s", target_dir, file_reldir);
-	const char* sce_filename = oscap_sprintf("%s/%s/%s", target_dir, file_reldir, file_basename);
-	ds_sds_dump_component(component_id, session, sce_filename, relative_filepath);
-	oscap_free(sce_filename);
+	const char* target_filename_dirname = oscap_sprintf("%s/%s",sub_dir, file_reldir);
 	oscap_free(filename_cpy);
 
+	ds_sds_dump_component(component_id, session, target_filename_dirname, relative_filepath);
+
```

Comment:
I don't agree.
When I write
```
printf("relative_filepath: %s| reldir %s \n", relative_filepath, file_reldir);
```
I will get
```
relative_filepath: xccdf.xml| reldir . 
relative_filepath: services/obsolete/disable_nfs_exports.sh| reldir ./services/obsolete 
```
So I expect, that your premise is wrong.







That path is ends with '.' isn't generally bad I think.
You can use **/home** or **/home/**, or when you build path from parts, you can get **/home/./././././** - strange, but as I know, should be always valid unix path.



Such building of paths already was in the code, but I tried to refactor it.
So If I didn't understand you well, please comment, we can improve it together.